### PR TITLE
Warn user on failure

### PR DESCRIPTION
**Pull request by karishmarajput**

### CODE REVIEW

**Review by joel122002 — APPROVED** [review]

**Comment by andinus** [comment]
You'll have to rebase this on top of latest master, it's out of date.

File: resources/app.js [hunk 1]
```diff
@@ -34,35 +36,47 @@ window.sendPaste = function sendPaste() {
...
-        paste.text = fflate.compressSync(buf, { level: 6, mem: 8 }).toString();
+        
+        try{
+            // Convert the compressed paste to string before storing it.
+            // Later we're stringifying this object, if we don't convert
+            // it to string here then JSON.stringify will preserve the
+            // Array structure, using much more space.
+            paste.text = fflate.compressSync(buf, { level: 6, mem: 8 }).toString();
+        }catch(err){
+            console.log("error: "+err);
```
Comment (andinus) [inline resources/app.js:47]: On compression failure, just disable compression and continue execution.

File: resources/app.js [hunk 2]
```diff
@@ -34,35 +36,47 @@ window.sendPaste = function sendPaste() {
...
-    noteAbovePasteBox.classList.remove("failure", "hidden");
-    noteAbovePasteBox.classList.add("success");
+        // Inform user about the Paste.
+        let link = document.createElement('a');
+        link.setAttribute('href', './#' + pasteUrl);
+        link.innerText = '#' + pasteUrl;
+        noteAbovePasteBox.appendChild(link);
 
+        noteAbovePasteBox.classList.remove("failure", "hidden");
+        noteAbovePasteBox.classList.add("success");
```
Comment (andinus) [inline resources/app.js:71]: These can be outside try block.

File: resources/app.js [hunk 3]
```diff
@@ -34,35 +36,47 @@ window.sendPaste = function sendPaste() {
...
+    } catch(err){
+        noteAbovePasteBox.innerText = "Encryption failed.";
+        pasteUrl="";
+        console.log("error"+ err);
+        noteAbovePasteBox.classList.remove("hidden");
+        noteAbovePasteBox.classList.add("failure");
```
Comment (andinus) [inline resources/app.js:77]: `return` early after adding these messages.

File: resources/app.js [hunk 4]
```diff
@@ -76,31 +90,39 @@ function initialize() {
...
+            // Inform user about the paste.
+            noteAbovePasteBox.innerText = "Paste successfully decrypted.";
+            noteAbovePasteBox.classList.remove("failure", "hidden");
+            noteAbovePasteBox.classList.add("success");
+        } catch(err){
+            console.log("error: "+ err);
+            outPaste="";
+            noteAbovePasteBox.innerText = "Paste not found.";
+            noteAbovePasteBox.classList.remove("hidden");
+            noteAbovePasteBox.classList.add("failure");
```
Comment (andinus) [inline resources/app.js:123]: Handle encryption and compression failure individually.

File: resources/app.js [hunk 5]
```diff
@@ -69,19 +83,36 @@ function initialize() {
...
             // Get the original Uint8Array generated by fflate by
             // converting paste.text to Array. It was converted to
             // string so the Array structure was not preserved.
-            const decompressed = fflate.decompressSync(
-                Uint8Array.from(paste.text.split(',').map(Number))
-            );
-
+            try{
+                const decompressed = fflate.decompressSync(
+                    Uint8Array.from(paste.text.split(',').map(Number))
```
Comment (andinus) [inline resources/app.js:107]: `decompressed` should be declared outside of try block, we're using it later.

**Comment by andinus** [comment]
Reverting the merge.

**Comment by andinus** [comment]
Reverted with https://github.com/noledgebin/frontend/commit/2d382cf79a96d33e3eff0824c0f639670ca36cac

**Pull request merged**